crates/oxc_diagnostics/src/lib.rs

```diff
@@ -100,6 +100,11 @@ impl Diagnostic for OxcDiagnostic {
             .map(Box::new)
             .map(|b| b as Box<dyn Iterator<Item = LabeledSpan>>)
     }
+
+    fn code<'a>(&'a self) -> Option<Box<dyn Display + 'a>> {
+        // self.code.is_some().then(|| Box::new(&self.code) as Box<dyn Display>)
+        None
```

Comment:
Right now, I'm rendering codes via `Display` for `OxcDiagnostic` for two reasons:
1. The graphical reporter renders error codes above error messages. (NOTE: do we want to change this behavior?)
2. Doing this lets stringification of diagnostics keep the error code.

Any opinions on how we should proceed?

Reply:
Or replace our `plugin-name(rule-name)` part with error code?

![CleanShot 2024-07-19 at 15 35 02@2x](https://github.com/user-attachments/assets/146b690d-597c-4b21-97bc-2f265fc38a76)

Reply:
cc: @Boshen